### PR TITLE
feat(@vtmn/css, @vtmn/react, @vtmn/svelte, @vtmn/vue): `divider`, add a11y requirements

### PR DESCRIPTION
**Pull request by GaspardMathon**
## Changes description

- `Divider` now has an `aria-labelledby` atribute, linked to its label's new span `id` attribute

## Context

- The `divider` didn't vocalize its label

## Checklist

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch**. Please, don't request directly from your main!
- [x] Check commits & PR names matches our requested structure. It must follow the https://www.conventionalcommits.org pattern.
- [x] Check your code additions will fail neither code linting checks.
- [x] I have reviewed the submitted code.
- [x] I have tested on related showcases.
- [x] If it includes design changes, please ask for a review with a core team designer.

## Does this introduce a breaking change?

- Yes, in order to be a11y compliant you need to nest the label into a span that has an `id` attribute as long as an `aria-labelledby` attribute on the divider div to allow vocalization of the label of the divider

### CODE REVIEW

File: packages/sources/svelte/src/components/structure/VtmnDivider/VtmnDivider.svelte [hunk 1]
```diff
@@ -34,9 +39,10 @@
   class={componentClass}
   role="separator"
   aria-orientation={orientation}
+  aria-labelledby={ariaLabelId}
   {...$$restProps}
 >
-  <slot />
+  <span id={ariaLabelId}><slot /></span>
```
Comment (Tlahey) [inline packages/sources/svelte/src/components/structure/VtmnDivider/VtmnDivider.svelte:45]: Why not keep `{...$$restProps}` to inject `aria-labelledby` ? 

```
  {...$$restProps}
>
  <span id={$$restProps['aria-labelledby']}><slot /></span>
```

**Comment by Tlahey** [comment]
@GaspardMathon It could be great to add a unit test on the slender side to check that the aria-labelby is applied to both levels of the component :)

File: packages/showcases/react/stories/components/structure/VtmnDivider/VtmnDivider.stories.tsx [hunk 2]
```diff
@@ -25,7 +25,7 @@ export default {
 
 const Template: Story = (args) => (
   <div style={{ width: 400, height: 250 }}>
-    <VtmnDivider {...args} />
+    <VtmnDivider aria-labelledby="title-example" {...args} />
```
Comment (lauthieb) [inline packages/showcases/react/stories/components/structure/VtmnDivider/VtmnDivider.stories.tsx:28]: And where is the element with the id `#title-example`? If you add this, you need to add this element in order to label the divider.

File: packages/showcases/svelte/stories/components/structure/VtmnDivider/VtmnDivider.stories.svelte [hunk 3]
```diff
@@ -27,6 +27,7 @@
 <Story class="block" name="Overview" let:args>
   <VtmnDivider
     class="story-divider-horizontal story-divider-vertical"
+    aria-labelledby="title-header-0"
```
Comment (lauthieb) [inline packages/showcases/svelte/stories/components/structure/VtmnDivider/VtmnDivider.stories.svelte:30]: And where is the element with the id `#title-header-0`? If you add this, you need to add this element in order to label the divider.

File: packages/showcases/svelte/stories/components/structure/VtmnDivider/VtmnDivider.stories.svelte [hunk 4]
```diff
@@ -39,17 +40,20 @@
     <VtmnDivider
       class="story-divider-horizontal"
       orientation="horizontal"
-      textPosition="start">Start label</VtmnDivider
+      textPosition="start"
+      aria-labelledby="title-header-1">Start label</VtmnDivider
```
Comment (lauthieb) [inline packages/showcases/svelte/stories/components/structure/VtmnDivider/VtmnDivider.stories.svelte:44]: Same remark for others

File: packages/showcases/vue/stories/components/structure/VtmnDivider/VtmnDivider.stories.js [hunk 5]
```diff
@@ -16,7 +16,7 @@ const Template = (args) => ({
   setup() {
     return { args };
   },
-  template: `<div style="width: 400px; height: 250px;"><VtmnDivider v-bind="args">Label</VtmnDivider></div>`,
+  template: `<div style="width: 400px; height: 250px;"><VtmnDivider v-bind="args" aria-labelledby="title-test">Label</VtmnDivider></div>`,
```
Comment (lauthieb) [inline packages/showcases/vue/stories/components/structure/VtmnDivider/VtmnDivider.stories.js:19]: And where is the element with the id `#title-test`? If you add this, you need to add this element in order to label the divider.

File: packages/sources/react/src/components/structure/VtmnDivider/VtmnDivider.tsx [hunk 6]
```diff
@@ -37,7 +44,7 @@ export const VtmnDivider = ({
       role="separator"
       aria-orientation={orientation}
     >
-      {children}
+      <span id={String(props['aria-labelledby'])}>{children}</span>
```
Comment (lauthieb) [inline packages/sources/react/src/components/structure/VtmnDivider/VtmnDivider.tsx:47]: We need to add this attribute only if `ariaLabelId` is defined.

File: packages/sources/svelte/src/components/structure/VtmnDivider/VtmnDivider.svelte [hunk 7]
```diff
@@ -36,7 +37,9 @@
   aria-orientation={orientation}
   {...$$restProps}
 >
-  <slot />
+  <span id={$$restProps['aria-labelledby']}>
```
Comment (lauthieb) [inline packages/sources/svelte/src/components/structure/VtmnDivider/VtmnDivider.svelte:40]: Why?

File: packages/sources/react/src/components/structure/VtmnDivider/VtmnDivider.tsx [hunk 8]
```diff
@@ -16,11 +16,18 @@ export interface VtmnDividerProps
    * @defaultValue 'start'
    */
   textPosition?: VtmnDividerTextPosition;
+
+  /**
+   * The position of the text inside the divider.
+   * @defaultValue undefined
+   */
+  ariaLabelId?: string;
```
Comment (lauthieb) [inline packages/sources/react/src/components/structure/VtmnDivider/VtmnDivider.tsx:24]: I would have prefered "ariaLabelledBy" instead of "ariaLabelId"

**Review by lauthieb — CHANGES_REQUESTED** [review]
Thanks @GaspardMathon, some remarks :)

**Comment by lauthieb** [comment]
I convert this to draft for now as there are several returns.

**Comment by GaspardMathon** [comment]
> 

done thanks :)

File: packages/sources/react/src/components/structure/VtmnDivider/VtmnDivider.tsx (hunk 6 above)
Comment (GaspardMathon) [reply packages/sources/react/src/components/structure/VtmnDivider/VtmnDivider.tsx]: Done, thanks :)

File: packages/showcases/svelte/stories/components/structure/VtmnDivider/VtmnDivider.stories.svelte (hunk 3 above)
Comment (lauthieb) [reply packages/showcases/svelte/stories/components/structure/VtmnDivider/VtmnDivider.stories.svelte:30]: Replace "aria-labelled-by" with "labelId"

Comment (GaspardMathon) [reply packages/showcases/svelte/stories/components/structure/VtmnDivider/VtmnDivider.stories.svelte:30]: I'm on it :)

**Review by lauthieb — CHANGES_REQUESTED** [review]

**Pull request merged**